code/game/objects/machinery/bots/bots.dm

```diff
@@ -1,18 +1,134 @@
+	var/stuck_counter = 0
+	///strings used on doing an action like cleaning
+	var/list/sentences = list()
+	///strings used on activating the bot
+	var/list/awakeningsentences = list()
+	///strings used on shutting down the robot
+	var/list/shutdownsentences = list()
+	///is the robot active
+	var/is_active = FALSE
+	///can the robot be turned on or off
+	var/alter_operating_mode = FALSE
+	///animation to play when the robot is started by hand
+	var/activation_animation = null
+	///animation to play when the robot is started by hand
+	var/deactivation_animation = null
+	///icon to set while active
+	var/active_icon_state = null
+
+/obj/machinery/bot/Initialize(mapload)
+	. = ..()
+	if(SStts.tts_enabled)
+		var/static/todays_voice
+		if(!todays_voice)
+			todays_voice = pick(SStts.available_speakers)
+		voice = todays_voice
+	RegisterSignal(src, COMSIG_AREA_EXITED, PROC_REF(turn_around))
+	if(is_active)
+		icon_state = active_icon_state
+		start_processing()
+
+/obj/machinery/bot/Destroy()
+	stop_processing()
+	return ..()
+
+///Turns the bot around when it leaves an area to make sure it doesnt wander off
+/obj/machinery/bot/proc/turn_around(datum/target)
+	SIGNAL_HANDLER
+	visible_message(span_warning("The [src] beeps angrily as it is moved out of it's designated area!"))
+	step_to(src, get_step(src,REVERSE_DIR(dir)))
+
+/obj/machinery/bot/process()
+	var/list/dirs = CARDINAL_DIRS - REVERSE_DIR(dir)
+	var/turf/selection
+	var/newdir
+	for(var/i=1 to length(dirs))
+		newdir = pick_n_take(dirs)
+		selection = get_step(src, newdir)
+		if(!selection.density)
+			break
+		newdir = null
+	if(!newdir)
+		say("DOOR STUCK, DOOOOR STUCK, AAAAAAH!")
+		return
+	step_to(src, get_step(src,newdir))
+
+/obj/machinery/bot/examine(mob/user, distance, infix, suffix)
+	. = ..()
+	. += "A panel on the top says it has cleaned [counter] items!"
+	. += "It is currently [is_active ? "on" : "off"]."
+
+/obj/machinery/bot/proc/reactivate()
+	stuck_counter = 0
+	if(!is_active) //no reactivation if somebody shut us off
+		return
+	start_processing()
+
+/obj/machinery/bot/Bump(atom/A)
+	. = ..()
+	if(++stuck_counter <= 3)
+		step_to(src, get_step(src, turn(dir, pick(90, -90))))
+		return
+	visible_message(span_warning("The [src] beeps angrily as it get stuck!"))
+	stop_processing()
+	addtimer(CALLBACK(src, PROC_REF(reactivate)), 20 SECONDS)
+
+/obj/machinery/bot/attack_hand(mob/living/user)
+	. = ..()
+	if(!ishuman(user))
+		return
+	if(!alter_operating_mode)
+		to_chat(user, "This robot doesn't have a switch.")
+		return
+	if(user.a_intent != INTENT_HELP)
+		return
+	switch(tgui_alert(user, "Do you you want to turn the [src] [is_active ? "off" : "on"]?" , "Cleanbot activation", list("No", "Yes")))
+		if("Yes")
+			if(is_active)
+				bot_shutdown(user)
+			else
+				bot_startup(user)
+
+/obj/machinery/bot/attack_ai(mob/user)
+	if(!alter_operating_mode)
+		to_chat(user, "This robot has a firewall and cannot be remotely accessed.")
+		return
+	if(is_active)
```

Comment:
Seems inconsistent with humans being asked for confirmation and AI toggling directly.